src/main/java/com/alex/eshop/service/CategoryService.java

```diff
@@ -16,16 +21,39 @@ public class CategoryService {
 
     private final CategoryMapper categoryMapper;
 
-    public CategoryService(CategoryRepository categoryRepository, CategoryMapper categoryMapper) {
+    private final CategoryUpdateMapper categoryUpdateMapper;
+
+    private final CategoryCreateMapper categoryCreateMapper;
+
+    public CategoryService(CategoryRepository categoryRepository, CategoryMapper categoryMapper,
+                           CategoryUpdateMapper categoryUpdateMapper, CategoryCreateMapper categoryCreateMapper) {
         this.categoryMapper = categoryMapper;
         this.categoryRepository = categoryRepository;
+        this.categoryUpdateMapper = categoryUpdateMapper;
+        this.categoryCreateMapper = categoryCreateMapper;
     }
 
     public List<CategoryDTO> getAllCategories() {
         return categoryMapper.toDto(categoryRepository.findAll());
     }
 
     public CategoryDTO getCategory(Long id) {
-        return categoryMapper.toDto(categoryRepository.findById(id).orElseThrow(()->new DataNotFound("There is no category with id " + id)));
+        return categoryMapper.toDto(categoryRepository.findById(id)
+                .orElseThrow(()->new DataNotFound("There is no category with id " + id)));
+    }
+    public Category createCategory(CategoryCreateDTO categoryCreateDTO){
+       return categoryRepository.save(categoryCreateMapper.toEntity(categoryCreateDTO));
+    }
+    public void updateCategory(CategoryUpdateDTO categoryUpdateDTO){
+        CategoryUpdateDTO categoryUpdate = categoryUpdateMapper
+                .toDto(categoryRepository.findById(categoryUpdateDTO.getId())
```

Comment:
For PUT operation there is not need to search existing category in db. Just update whole entity as is.
You can read about difference between PUT and PATCH methods. For PATCH you will need to search in DB, for PUT - not.